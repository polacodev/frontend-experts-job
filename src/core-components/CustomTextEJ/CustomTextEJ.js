import React from 'react';
import { Text } from 'native-base';

import styles from './CustomTextEJ.style';

const CustomTextEJ = ({ type = '', children, color, size }) => {
  const typeOfText = (element) => {
    switch (element) {
      case 'normal':
        return (
          <Text numberOfLines={1} style={styles(color, size).customText}>
            {children}
          </Text>
        );
      case 'note':
        return (
          <Text note numberOfLines={1} style={styles(color, size).customText}>
            {children}
          </Text>
        );
      default:
        return null;
    }
  };

  return typeOfText(type);
};

export default CustomTextEJ;
