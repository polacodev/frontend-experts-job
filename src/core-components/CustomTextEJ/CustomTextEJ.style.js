import { StyleSheet } from 'react-native';

export const styles = (color, size) =>
  StyleSheet.create({
    customText: {
      fontSize: size,
      color: color,
      paddingTop: 2,
    },
  });

export default styles;
