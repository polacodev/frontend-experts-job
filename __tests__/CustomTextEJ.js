import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';

import CustomTextEJ from '../src/core-components/CustomTextEJ/CustomTextEJ';
jest.useFakeTimers();
describe('<CustomTextEJ />', () => {
  test('renders <CustomTextEJ /> correctly', () => {
    const tree = renderer
      .create(
        <CustomTextEJ
          type="normal"
          children="test label normal"
          color="#ffffff"
          size={20}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should find "test label normal"', async () => {
    const { getByText } = render(
      <CustomTextEJ
        type="note"
        children="test label normal"
        color="#ffffff"
        size={20}
      />,
    );
    expect(getByText('test label normal').children).toBeDefined();
  });

  test('should find "test label note"', async () => {
    const { getByText } = render(
      <CustomTextEJ
        type="note"
        children="test label note"
        color="#ffffff"
        size={20}
      />,
    );
    expect(getByText('test label note').children).toBeDefined();
  });

  test('should find "test label empty"', async () => {
    const { container } = render(
      <CustomTextEJ
        type="lol"
        children="test label empty"
        color="#ffffff"
        size={20}
      />,
    );
    expect(container.firstChild).toBeUndefined();
  });

  test('should return undefined due the empty props', async () => {
    const { container } = render(<CustomTextEJ />);
    expect(container.firstChild).toBeUndefined();
  });
});
