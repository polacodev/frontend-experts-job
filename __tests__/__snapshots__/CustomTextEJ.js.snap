// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTextEJ /> renders <CustomTextEJ /> correctly 1`] = `
<Text
  numberOfLines={1}
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      },
      Object {
        "color": "#ffffff",
        "fontSize": 20,
        "paddingTop": 2,
      },
    ]
  }
  uppercase={false}
>
  test label normal
</Text>
`;
